default/decorators.py

```diff
@@ -41,3 +41,42 @@ def wrap(request, *args, **kwargs):
         return function(request, *args, **kwargs)
 
     return wrap
+
+
+def split_size(function):
+    @wraps(function)
+    def wrap(request, *args, **kwargs):
+        split_size = request.GET.get('splitSize')
+        if split_size.isdecimal():
+            kwargs['size'] = int(split_size)
+        else:
+            msg = _('An invalid split size was submitted, and therefore ignored. Default value is used, split size: 1')
+            if 'warnings' in kwargs:
+                kwargs['warnings'].append(msg)
+            else:
+                kwargs['warnings'] = [msg]
+        return function(request, *args, **kwargs)
+
+    return wrap
+
+
+def optional_args(function):
+    @wraps(function)
+    def wrap(request, *args, **kwargs):
+        encoding = request.GET.get('encoding', 'utf-8')
+        try:
+            teststr = "test"
+            teststr.encode(encoding)
+            kwargs['encoding'] = encoding
+        except LookupError:
+            msg = _('Encoding %(encoding)s ... is not recognized. The default value \'utf-8\' was used.') % {
+                        'encoding': encoding[0:64],
```

Comment:
Do we know why a substring (`encoding[0:64]`) is used?

Reply:
I will ask.

Reply:
Then the input should be validated, not just cut out, don't you agree?

Reply:
Validated in `optional_arguments.html` with `maxlength`.